FlowCrypt/Extensions/URLSessionExtension.swift

```diff
@@ -56,6 +56,35 @@ extension URLSession {
     }
 }
 
+extension URLSession {
+    func asyncCall(_ urlRequest: URLRequest, tolerateStatus: [Int]? = nil) async throws -> HttpRes {
+        let trace = Trace(id: "call")
+        let (data, response) = try await self.data(for: urlRequest)
```

Comment:
Can you write a test to confirm that this will not immediately throw on a 404 or unreachable server? The expected behavior is that it should continue below and throw a `HttpErr` - that's what it did before.

Reply:
I changed code to handle any error from `data()` method internally and map it to `HttpErr`. If we have network connection and server is unreachable we will get 404 but if we don't have network connection we will get `HttpErr` wrapped network error.